### PR TITLE
Adds Plant Fiber and products.

### PR DESCRIPTION
**Pull request by nullbear**
:cl:
add: Added plant fiber/cellulose. It can be used to make cloth, cardboard, and paper products, or reprocessed into sugar and/or oil. It can obtained from fibrous plants such as cannabis and wood. 
add: Added paper and cardboard recipes to the biogenerator.
/:cl:

Adds plant fiber + products. A method of obtaining cardboard, cloth, and
paper, using hydroponics and chemistry.

Adds plant fiber to a couple of 'fibrous plants'.
Makes wood and cardboard give fiber instead of carbon when processed in
the chemgrinder.

Melting the fiber in a strong acid (with water) converts it to sugar.

Heating the fiber converts it to oil, and subsequently to ash. This can
be mixed with water to make lye, one of the ingredients used in all of
the plant-product recipes.

All of the plant-products are processed using fiber, lye, and
pestkiller. Paper is produced by adding drying_agent, cardboard by
cooking it, and cloth by adding sulfuric acid to the mixture.

Adds a safety/sanity check to stacks, preventing stacks from being
created with more than their max_amount, instead splitting the amount
across several stacks.

Plant fiber is also useful for gaining biogenerator points.

Only thing this needs to be fully accessible to hydroponics, is some
ghetto chem equipment for heating and grinding up chems, which will
probably be added in a later PR.

### CODE REVIEW

**Comment by ma44** [comment]
Idk how this would be cause im job banned but more bio points sound fun

File: code/game/objects/items/stacks/stack.dm [hunk 1]
```diff
@@ -22,7 +22,11 @@
 /obj/item/stack/New(var/loc, var/amount=null)
 	..()
 	if (amount)
-		src.amount = amount
+		if(amount<=max_amount)
+			src.amount = amount
+		else
+			amount = max_amount
```
Comment (phil235) [inline code/game/objects/items/stacks/stack.dm]: `src.amount = max_amount` , having an arg and a variable of the src use the same name was a shit idea.

File: code/modules/reagents/chemistry/recipes/others.dm [hunk 2]
```diff
@@ -116,6 +116,45 @@
...
+	id = "cardification"
+	result = null
+	required_reagents = list("plantfiber" = 10, "lye" = 5, "pestkiller" = 5)
+	required_temp = 374
+	result_amount = 1
+
+/datum/chemical_reaction/cardification/on_reaction(datum/reagents/holder, created_volume)
+	var/location = get_turf(holder.my_atom)
+	new /obj/item/stack/sheet/cardboard(location, created_volume)
+	return
```
Comment (phil235) [inline code/modules/reagents/chemistry/recipes/others.dm]: you don't need to add a return at the end of the proc.

File: code/modules/reagents/chemistry/recipes/others.dm [hunk 3]
```diff
@@ -116,6 +116,45 @@
...
+	result = null
+	required_reagents = list("plantfiber" = 2, "lye" = 1, "pestkiller" = 1)
+	required_catalysts = list("drying_agent" = 1)
+	result_amount = 1
+
+/datum/chemical_reaction/paperification/on_reaction(datum/reagents/holder, created_volume)
+	var/location = get_turf(holder.my_atom)
+	for(var/i = 1, i <= created_volume, i++)
+		new /obj/item/weapon/paper(location)
+	return
```
Comment (phil235) [inline code/modules/reagents/chemistry/recipes/others.dm]: return

File: code/modules/reagents/chemistry/recipes/others.dm (hunk 2 above)
Comment (nullbear) [inline code/modules/reagents/chemistry/recipes/others.dm]: ah, i saw it in the meatification one so i assumed.

**Comment by Core0verload** [comment]
Why you do this instead of simply extending the biomass? It serves exactly the same purpose of "material from plants".

**Comment by nullbear** [comment]
I consider biomass to be 'cellular material' while this is a chemical 'C6H12O5'
As well, i did not see it anywhere, what's it used for?

**Comment by nullbear** [comment]
As in this fiber could be made from biomass. But not vice versa. Biomass could include dna and hair and even living organisms.

**Comment by Core0verload** [comment]
Biogenerator. It converts plants directly to biomass, and then uses biomass to make clothing or reagents that are usually made from organic materials. Exactly the same purpose as your "plant fiber", and useable both by botany and random assistants.

It has one benefit over your system: it's not an another feature nobody would ever use.

**Comment by nullbear** [comment]
I should add a recipe for the biogenerator to convert biomass into plant fiber though. If that's what you meant.

**Comment by Core0verload** [comment]
No. I meant this would better be +4 recipes to biogen instead of +4 ultra special chem recipes nobody would ever use.

**Comment by nullbear** [comment]
My system serves the same purpose as synthflesh. An alternate method of obtaining the same thing. The biogenerator should have these as built in recipes, though.

**Comment by nullbear** [comment]
I agree.

**Comment by Core0verload** [comment]
Still not worth having over biogen. Synthflesh at least has medical uses.

**Comment by nullbear** [comment]
my headcannon is that the biogenerator's biomass is primarily cellulose/plant fiber, a percentage of 'trace' chemicals, and water. The machine itself is kind of a hybrid between the chemistry setup, and an autolathe, which takes all the complexity out of both, in exchange for more limited features, and reduced chemical efficiency.

If you were to do the stuff the biogenerator does, without the biogenerator, it'd essentially be grinding up the individual plants into their constituent compounds, and doing the whole chemistry thing some more to put them back together as milk/cloth/etc. Then you've got the protolathe bit that turns the materials into items. 

It makes sense to include both. It shows consistency and material persistence, as well as provides alternative methods of reaching the same goal. In this case, the goal is pretty pathetic (you get even MORE paper and cardboard. Wheee!!) but whatever.

**Comment by Core0verload** [comment]
My headcannon is that biogenerator's biomass is trash plants waiting to get turned into something useful. Do not add pointless complexity. Do not overthink.

**Comment by Delimusca** [comment]
@Core0verload 
botany has no complexity, and having more chemistry outside the hardcoded chemlab is good.

**Comment by PKPenguin321** [comment]
>botany has no complexity

**Comment by ExcessiveUseOfCobblestone** [comment]
Put this under the biogen so it'll benefit from science upgrades [do people do this considering the need for botany involvement?].

**Comment by nullbear** [comment]
I've added it under both. The biogen can create the paper recipes from biomass. And so can chemistry. The material/chemical costs are about equal for either recipe, with chemistry being more efficient than a stock biogenerator. The fiber->sugar and fiber->oil->ash conversion recipes are still included under chemistry, and are inaccessible to hydro until a ghetto chem heater is added.

**Comment by Bawhoppen** [comment]
why can't you just make a few bulletpoints instead of a wall of text that I'm not going to read so now I have to thumbs-down

**Comment by nullbear** [comment]
Bawhoppen there are literally 2 bullets in total.

**Comment by Bawhoppen** [comment]
it looks like a wall of text explanation to me

**Comment by nullbear** [comment]
To explain every single change i made to the code, for the reviewer/s.

File: code/game/objects/items/stacks/stack.dm [hunk 4]
```diff
@@ -22,7 +22,11 @@
 /obj/item/stack/New(var/loc, var/amount=null)
 	..()
 	if (amount)
-		src.amount = amount
+		if(amount<=max_amount)
```
Comment (optimumtact) [inline code/game/objects/items/stacks/stack.dm]: put spaces 
   amount <= max_amount

File: code/game/objects/items/stacks/stack.dm [hunk 5]
```diff
@@ -22,7 +22,11 @@
...
+			src.amount = amount
```
Comment (optimumtact) [inline code/game/objects/items/stacks/stack.dm]: remove src's, they're only used when there is an actual need

File: code/modules/hydroponics/biogenerator.dm [hunk 6]
```diff
@@ -213,9 +213,9 @@
 	var/S = 0
 	for(var/obj/item/weapon/reagent_containers/food/snacks/grown/I in contents)
 		S += 5
-		if(I.reagents.get_reagent_amount("nutriment") < 0.1)
+		if((I.reagents.get_reagent_amount("nutriment")+I.reagents.get_reagent_amount("plantfiber")) < 0.1)
 			points += 1*productivity
-		else points += I.reagents.get_reagent_amount("nutriment")*10*productivity
+		else points += (I.reagents.get_reagent_amount("nutriment")+I.reagents.get_reagent_amount("plantfiber"))*10*productivity
```
Comment (optimumtact) [inline code/modules/hydroponics/biogenerator.dm]: spaces around the +

**Comment by nullbear** [comment]
closing this until after my chemistry refactor.

File: code/modules/hydroponics/biogenerator.dm (hunk 6 above)
Comment (nullbear) [inline code/modules/hydroponics/biogenerator.dm]: is that actually a thing? :O

File: code/game/objects/items/stacks/stack.dm (hunk 5 above)
Comment (nullbear) [inline code/game/objects/items/stacks/stack.dm]: new is called with a variable called amount. src is used to set src amount to the amount argument. Isn't this an actual need for it?

Comment (optimumtact) [inline code/game/objects/items/stacks/stack.dm]: oh I didn't spot that sorry, yeah looks fine

**Pull request closed without merging**